lib/board.rb

```diff
@@ -1,11 +1,23 @@
 class Board
-  attr_reader :board
+  attr_reader :board, :moves_remaining
 
   def initialize(board)
     @board = board
+    @moves_remaining = 9
   end
 
   def player_make_move(player_mark, position)
     @board[position - 1] = player_mark
+    update_moves_remaining
+  end
+
+  def position_available?(position)
+    @board[position - 1].is_a? Integer
+  end
+
+  private
+  def update_moves_remaining
+    update_moves = @board.count { |position| position.is_a? Integer }
```

Comment:
Is `@moves_remaining` necessary? 

What do you think about a design where each time you need to check if there are moves remaining, you call `@board.count { |position| position.is_a? Integer }`, without storing the number of `@moves_remaining` on `Board`?

Reply:
Good point! Getting rid of `@moves_remaining` in board also means I can remove the `moves_remaining?` method in game. Thanks, Justyna

Reply:
@itsellej yep, this is exactly what I meant! :)